src/std/misc/list.ss

```diff
@@ -0,0 +1,88 @@
+;; -*- Gerbil -*-
+package: std/misc
+;;;; List utilities
+
+(export
+  plist->alist
+  length=? length=n?
+  length<? length<n? length<=? length<=n?
+  length>? length>n? length>=? length>=n?
+  call-with-list-builder
+  snoc append1)
+
+(import
+  :std/srfi/1
+  :utils/base)
+
+;; This function transform a property list (k1 v1 k2 v2 ...) into
+;; an association list ((k1 . v1) (k2 . v2) ...).
+;; NB: it uses equal? as the equality predicate for keys.
+(def (plist->alist plist)
+  (hash->list (plist->hash-table plist)))
```

Comment:
this is not structure preserving, and also inefficient.
Can you write this with a simple recursive converter?

Reply:
Probably. I admit I also wanted to normalize in case of duplicate (shadowed) key, but maybe that's the wrong place for that.

Reply:
Fixing...